mono_repo/lib/src/commands/pub.dart

```diff
@@ -5,116 +5,291 @@
 import 'dart:async';
 import 'dart:io';
 
+import 'package:args/args.dart';
 import 'package:args/command_runner.dart';
 import 'package:io/ansi.dart';
-import 'package:meta/meta.dart';
 import 'package:path/path.dart' as p;
 
 import '../root_config.dart';
 import 'mono_repo_command.dart';
 
-class PubCommand extends Command<void> {
-  PubCommand() {
-    addSubcommand(_PubSubCommand('get'));
-    addSubcommand(_PubSubCommand('upgrade'));
-  }
+class PubCommand extends MonoRepoCommand {
+  @override
+  ArgParser get argParser => _pubArgParser ??= _PubArgParser();
+  _PubArgParser _pubArgParser;
+
+  @override
+  String get description => 'Run a `pub` command across all packages.';
 
   @override
   String get name => 'pub';
 
   @override
-  String get description =>
-      'Run `pub get` or `pub upgrade` against all packages.';
+  List<String> get aliases => ['flutter'];
+
+  @override
+  Future<void> run() => pub(rootConfig(), argResults.rest);
 }
 
-const _offline = 'offline';
-const _dryRun = 'dry-run';
-const _precompile = 'precompile';
-
-class _PubSubCommand extends MonoRepoCommand {
-  @override
-  final String name;
-
-  _PubSubCommand(this.name) {
-    argParser
-      ..addFlag(_offline,
-          negatable: true,
-          defaultsTo: false,
-          help: 'Use cached packages instead of accessing the network.')
-      ..addFlag(_dryRun,
-          abbr: 'n',
-          defaultsTo: false,
-          negatable: false,
-          help: 'Precompile executables and transformed dependencies.')
-      ..addFlag(_precompile,
-          defaultsTo: true,
-          negatable: true,
-          help: "Report what dependencies would change but don't change any.");
-  }
+/// Implementation of [ArgParser] that delegates to an [allowAnything()] but
+/// allows compatibility with [Command] and [CommandRunner].
+///
+/// Specifically, the follow issues are addressed:
+/// - [Command]:
+///   - Attempts to [addFlag] a 'help' option, which isn't allowed by
+///   [allowAnything()] (ironically). We don't want that flag to be added
+///   anyway; we want `pub` or `flutter` to handle printing their own usage.
+///   [allowAnything()] throws an [UnsupportedError] so instead we just
+///   ignore the call and don't pass it on to the delegate.
+/// - [CommandRunner]:
+///   - Attempts to check the value of the 'help' option, which doesn't exist
+///   because we ignored [Command]'s adding it. This check isn't done via
+///   [_PubArgParser]'s own [ArgResults] from [parse], as this isn't ever the
+///   top-level [ArgParser]. Instead, as [CommandRunner] steps into the
+///   arguments for [PubCommand], it ends up being the [options] getter here
+///   which is checked for the 'help' option by an [ArgResults] created from
+///   the top-level (which is out of our control). To trick [CommandRunner],
+///   we return a [_PubOptions], itself a delegating class.
+class _PubArgParser implements ArgParser {
```

Comment:
Implementing this is also a bit scary, we would probably want a tight version constraint on `args` if we do this (with an upper bound < the next minor version instead of major)